File: dev-c++/1822-sign-of-the-product-of-an-array.cpp
class Solution {
public:
    int arraySign(vector<int>& nums) {
        bool is_positive = true;

        for(auto n: nums) {
            if(n == 0) return 0;
            if(n <  0) { is_positive = !is_positive; }
        }

        return (is_positive) ? 1 : -1;
    }
};

File: dev-rust/1822-sign-of-the-product-of-an-array.rs
impl Solution {

    pub fn array_sign(nums: Vec<i32>) -> i32 {
        let mut negative = false;

        for n in nums {
            if n == 0 { return 0; }
            if n <  0 { negative = !negative; }
        }

        return if negative { -1 } else { 1 };
    }
}